downgrade

```diff
@@ -103,8 +113,8 @@ main() {
   for term; do
     candidates=( $(search_packages "$term" | sort_packages) )
 
-    if present_packages "${candidates[@]}"; then
-      if choice="$(read_selection "${candidates[@]}")"; then
+    if check_packages || test "${#candidates[@]}" -gt 0; then
+      if check_selection || choice="${candidates[0]}"; then
```

Comment:
check_packages executes the function present_packages only if there is more than one possibility to download (e.g. if you search a incomplete version like 2.3 being versions 2.3-1 or 2.3-2 since the search of the version is a grep). Then, if there is more than one version (test "${#candidates[@]}" -gt 0) and after being potentially executed present_packages, it continues with the program.
check_selection executes the function read_selection only if there is more than one possibility to download (the same case than above). If there is only one possibility, then the choice is that version and it is not needed to ask the user the number of the version to download (check_selection || choice="${candidates[0]}")